[ffmpeg:arm64-ios] build failure for ios

**Host Environment**
- OS: macOS Monterey 12.2 Beta
- Compiler: GNU Make 3.81

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install ffmpeg:arm64-ios`

**Failure logs**
-Cut and paste the appropriate build messages from the console output.
```
Computing installation plan...
The following packages will be built and installed:
    ffmpeg[avcodec,avdevice,avfilter,avformat,core,swresample,swscale]:arm64-ios -> 4.4.1#13
Detecting compiler hash for triplet arm64-ios...
Restored 0 packages from /../.cache/vcpkg/archives in 20.18 us. Use --debug to see more details.
Installing 1/1 ffmpeg:arm64-ios...
Building ffmpeg[avcodec,avdevice,avfilter,avformat,core,swresample,swscale]:arm64-ios...
-- Using community triplet arm64-ios. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: /../vcpkg/triplets/community/arm64-ios.cmake
-- Using cached ffmpeg-ffmpeg-n4.4.1.tar.gz.
-- Cleaning sources at /../vcpkg/buildtrees/ffmpeg/src/n4.4.1-136f30bfcb.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /../vcpkg/downloads/ffmpeg-ffmpeg-n4.4.1.tar.gz
-- Applying patch 0001-create-lib-libraries.patch
-- Applying patch 0003-fix-windowsinclude.patch
-- Applying patch 0004-fix-debug-build.patch
-- Applying patch 0006-fix-StaticFeatures.patch
-- Applying patch 0007-fix-lib-naming.patch
-- Applying patch 0009-Fix-fdk-detection.patch
-- Applying patch 0010-Fix-x264-detection.patch
-- Applying patch 0011-Fix-x265-detection.patch
-- Applying patch 0012-Fix-ssl-110-detection.patch
-- Applying patch 0013-define-WINVER.patch
-- Applying patch 0014-avfilter-dependency-fix.patch
-- Applying patch 0015-Fix-xml2-detection.patch
-- Applying patch 0016-configure-dnn-needs-avformat.patch
-- Applying patch 0018-libaom-Dont-use-aom_codec_av1_dx_algo.patch
-- Applying patch 0019-libx264-Do-not-explicitly-set-X264_API_IMPORTS.patch
-- Using source at /../vcpkg/buildtrees/ffmpeg/src/n4.4.1-136f30bfcb.clean
-- Getting CMake variables for arm64-ios-dbg
-- Getting CMake variables for arm64-ios-rel
-- Building Options: --enable-pic --disable-doc --enable-debug --enable-runtime-cpudetect --enable-asm --disable-x86asm --disable-ffmpeg --disable-ffplay --disable-ffprobe --disable-alsa --enable-avcodec --enable-avdevice --enable-avformat --enable-avfilter --disable-postproc --enable-swresample --enable-swscale --disable-libaom --disable-libass --disable-avisynth --disable-bzlib --disable-libdav1d --disable-libfdk-aac --disable-libfontconfig --disable-libfreetype --disable-libfribidi --disable-iconv --disable-libilbc --disable-lzma --disable-libmp3lame --disable-libmodplug --disable-cuda --disable-nvenc --disable-nvdec  --disable-cuvid --disable-ffnvcodec --disable-opencl --disable-opengl --disable-libopenh264 --disable-libopenjpeg --disable-openssl --disable-libopus --disable-sdl2 --disable-libsnappy --disable-libsoxr --disable-libspeex --disable-libssh --disable-libtensorflow --disable-libtesseract --disable-libtheora --disable-libvorbis --disable-libvpx --disable-libwebp --disable-libx264 --disable-libx265 --disable-libxml2 --disable-zlib  --enable-cross-compile --arch=arm64 --pkg-config-flags=--static
-- Building Release Options:
-- Building ffmpeg for Release
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:128 (message):
    Command failed: /bin/sh ./build.sh
    Working Directory: /../vcpkg/buildtrees/ffmpeg/arm64-ios-rel
    Error code: 2
    See logs for more information:
      /../vcpkg/buildtrees/ffmpeg/build-arm64-ios-rel-out.log
      /..vcpkg/buildtrees/ffmpeg/build-arm64-ios-rel-err.log

Call Stack (most recent call first):
  ports/ffmpeg/portfile.cmake:560 (vcpkg_execute_required_process)
  scripts/ports.cmake:146 (include)
```
-Please attach any additional failure logs mentioned in the console output.
Error in build-arm64-ios-rel-err.log:
```
src/libavdevice/audiotoolbox.m:78:5: error: unknown type name 'AudioDeviceID'; did you mean 'AudioFileID'?
    AudioDeviceID *devices;
    ^~~~~~~~~~~~~
    AudioFileID
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS15.0.sdk/System/Library/Frameworks/AudioToolbox.framework/Headers/AudioFile.h:197:35: note: 'AudioFileID' declared here
typedef struct OpaqueAudioFileID        *AudioFileID;
                                         ^
src/libavdevice/audiotoolbox.m:84:5: error: use of undeclared identifier 'AudioObjectPropertyAddress'
    AudioObjectPropertyAddress prop;
    ^
src/libavdevice/audiotoolbox.m:85:5: error: use of undeclared identifier 'prop'
    prop.mSelector = kAudioHardwarePropertyDevices;
    ^
src/libavdevice/audiotoolbox.m:85:22: error: use of undeclared identifier 'kAudioHardwarePropertyDevices'
    prop.mSelector = kAudioHardwarePropertyDevices;
                     ^
src/libavdevice/audiotoolbox.m:86:5: error: use of undeclared identifier 'prop'
    prop.mScope    = kAudioObjectPropertyScopeGlobal;
    ^
src/libavdevice/audiotoolbox.m:86:22: error: use of undeclared identifier 'kAudioObjectPropertyScopeGlobal'
    prop.mScope    = kAudioObjectPropertyScopeGlobal;
                     ^
src/libavdevice/audiotoolbox.m:87:5: error: use of undeclared identifier 'prop'
    prop.mElement  = kAudioObjectPropertyElementMaster;
    ^
src/libavdevice/audiotoolbox.m:87:22: error: use of undeclared identifier 'kAudioObjectPropertyElementMaster'; did you mean 'kAudioUnitProperty_ElementName'?
    prop.mElement  = kAudioObjectPropertyElementMaster;
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                     kAudioUnitProperty_ElementName
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS15.0.sdk/System/Library/Frameworks/AudioToolbox.framework/Headers/AudioUnitProperties.h:876:2: note: 'kAudioUnitProperty_ElementName' declared here
        kAudioUnitProperty_ElementName                                  = 30,
        ^
src/libavdevice/audiotoolbox.m:88:11: error: implicit declaration of function 'AudioObjectGetPropertyDataSize' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    err = AudioObjectGetPropertyDataSize(kAudioObjectSystemObject, &prop, 0, NULL, &data_size);
          ^
src/libavdevice/audiotoolbox.m:88:11: note: did you mean 'AudioQueueGetPropertySize'?
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS15.0.sdk/System/Library/Frameworks/AudioToolbox.framework/Headers/AudioQueue.h:1458:1: note: 'AudioQueueGetPropertySize' declared here
AudioQueueGetPropertySize(          AudioQueueRef           inAQ,
^
src/libavdevice/audiotoolbox.m:88:42: error: use of undeclared identifier 'kAudioObjectSystemObject'
    err = AudioObjectGetPropertyDataSize(kAudioObjectSystemObject, &prop, 0, NULL, &data_size);
                                         ^
src/libavdevice/audiotoolbox.m:88:69: error: use of undeclared identifier 'prop'
    err = AudioObjectGetPropertyDataSize(kAudioObjectSystemObject, &prop, 0, NULL, &data_size);
                                                                    ^
src/libavdevice/audiotoolbox.m:92:38: error: use of undeclared identifier 'AudioDeviceID'
    num_devices = data_size / sizeof(AudioDeviceID);
                                     ^
src/libavdevice/audiotoolbox.m:94:30: error: expected expression
    devices = (AudioDeviceID*)(av_malloc(data_size));
                             ^
src/libavdevice/audiotoolbox.m:94:16: error: use of undeclared identifier 'AudioDeviceID'
    devices = (AudioDeviceID*)(av_malloc(data_size));
               ^
src/libavdevice/audiotoolbox.m:95:11: error: implicit declaration of function 'AudioObjectGetPropertyData' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    err = AudioObjectGetPropertyData(kAudioObjectSystemObject, &prop, 0, NULL, &data_size, devices);
          ^
src/libavdevice/audiotoolbox.m:95:11: note: did you mean 'AudioObjectGetPropertyDataSize'?
src/libavdevice/audiotoolbox.m:88:11: note: 'AudioObjectGetPropertyDataSize' declared here
    err = AudioObjectGetPropertyDataSize(kAudioObjectSystemObject, &prop, 0, NULL, &data_size);
          ^
src/libavdevice/audiotoolbox.m:95:38: error: use of undeclared identifier 'kAudioObjectSystemObject'
    err = AudioObjectGetPropertyData(kAudioObjectSystemObject, &prop, 0, NULL, &data_size, devices);
                                     ^
src/libavdevice/audiotoolbox.m:95:65: error: use of undeclared identifier 'prop'
    err = AudioObjectGetPropertyData(kAudioObjectSystemObject, &prop, 0, NULL, &data_size, devices);
```

**Additional context**
Issue resolved after I added the following code to ports/ffmpeg/profile.cmake
```
if (VCPKG_TARGET_IS_IOS)
    set(OPTIONS "${OPTIONS} --disable-audiotoolbox")
endif()
```

--- a/ports/ffmpeg/portfile.cmake
+++ b/ports/ffmpeg/portfile.cmake
@@ -460,6 +460,10 @@ if (VCPKG_TARGET_IS_OSX)
     set(OPTIONS "${OPTIONS} --disable-vdpau") # disable vdpau in OSX
 endif()
 
+if(VCPKG_TARGET_IS_IOS)
+    set(OPTIONS "${OPTIONS} --disable-audiotoolbox") # disable AudioToolbox on iOS
+endif()
+
 set(OPTIONS_CROSS " --enable-cross-compile")
 
 # ffmpeg needs --cross-prefix option to use appropriate tools for cross-compiling.

--- a/ports/ffmpeg/vcpkg.json
+++ b/ports/ffmpeg/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "ffmpeg",
   "version": "4.4.1",
-  "port-version": 18,
+  "port-version": 19,
   "description": [
     "a library to decode, encode, transcode, mux, demux, stream, filter and play pretty much anything that humans and machines have created.",
     "FFmpeg is the leading multimedia framework, able to decode, encode, transcode, mux, demux, stream, filter and play pretty much anything that humans and machines have created. It supports the most obscure ancient formats up to the cutting edge. No matter if they were designed by some standards committee, the community or a corporation. It is also highly portable: FFmpeg compiles, runs, and passes our testing infrastructure FATE across Linux, Mac OS X, Microsoft Windows, the BSDs, Solaris, etc. under a wide variety of build environments, machine architectures, and configurations."

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2278,7 +2278,7 @@
     },
     "ffmpeg": {
       "baseline": "4.4.1",
-      "port-version": 18
+      "port-version": 19
     },
     "ffnvcodec": {
       "baseline": "11.1.5.0",

--- a/versions/f-/ffmpeg.json
+++ b/versions/f-/ffmpeg.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "0ba5a87e25f9b9f3bf253a559830b829542c6bd5",
+      "version": "4.4.1",
+      "port-version": 19
+    },
+    {
       "git-tree": "65fe58cd3c36c19b6ad4b104b9ff57c506009e9a",
       "version": "4.4.1",
       "port-version": 18

--- a/versions/f-/ffmpeg.json
+++ b/versions/f-/ffmpeg.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "0ba5a87e25f9b9f3bf253a559830b829542c6bd5",
+      "git-tree": "81adf5d3e80f0e54a7e968950860fed901186e85",
       "version": "4.4.1",
       "port-version": 19
     },

